### PR TITLE
Java: Add local version of the XXE query

### PR DESCRIPTION
**Pull request by atorralba**
Adds a version of the XXE query that uses local sources. This helps to cover (at least) the following CVEs:
* CVE-2020-2138
* CVE-2021-21701

### CODE REVIEW

File: java/ql/lib/semmle/code/java/security/XxeQuery.qll [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/** Provides taint tracking configurations to be used in XXE queries. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.TaintTracking2
+import semmle.code.java.security.XmlParsers
+import DataFlow::PathGraph
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/XxeQuery.qll:8]: Let's not put top-level `query` predicates outside `.ql` files, I think.
```suggestion
```

File: java/ql/lib/semmle/code/java/security/XxeQuery.qll [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  override predicate isSource(DataFlow::Node src) { src instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof UnsafeXxeSink }
+}
+
+/**
+ * A taint-tracking configuration for unvalidated local user input that is used in XML external entity expansion.
+ */
+class XxeLocalConfig extends TaintTracking::Configuration {
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/XxeQuery.qll:24]: Hold, wait! You can't put both of these configurations in the same file - that'll ruin performance of the regular XXE query,

**Review by aschackmull — CHANGES_REQUESTED** [review]
We may soon be able to put two configurations in a single file like this once the parameterisation rewrite lands, but not just yet.

File: java/ql/lib/semmle/code/java/security/XxeQuery.qll (hunk 1 above)
Comment (atorralba) [reply java/ql/lib/semmle/code/java/security/XxeQuery.qll]: Yeah, copy paste error, thank you and Code Scanning for spotting it!

File: java/ql/lib/semmle/code/java/security/XxeQuery.qll (hunk 2 above)
Comment (atorralba) [reply java/ql/lib/semmle/code/java/security/XxeQuery.qll:23]: 🤦‍♂️ That was a silly mistake, sorry. I'll split that up.

File: java/ql/lib/semmle/code/java/security/Xxe.qll [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/** Provides classes to reason about XML eXternal Entity (XXE) vulnerabilities. */
+
+import java
+import semmle.code.java.security.XmlParsers
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/security/Xxe.qll:4]: unused import

Comment (aschackmull) [reply java/ql/lib/semmle/code/java/security/Xxe.qll:4]: Or, I guess it's probably used in `java/ql/lib/semmle/code/java/security/XxeQuery.qll`, so move it there?

**Review by aschackmull — APPROVED** [review]

**Pull request merged**